### PR TITLE
Changed to throw IllegalStateException when setAutoRefresh() is called on non-Looper thread

### PR DESCRIPTION
**Pull request by zaki50**
@realm/java

### CODE REVIEW

File: realm/realm-library/src/main/java/io/realm/BaseRealm.java [hunk 1]
```diff
@@ -89,7 +89,7 @@ protected BaseRealm(RealmConfiguration configuration, boolean autoRefresh) {
      * thread.
      *
      * @param autoRefresh {@code true} will turn auto-refresh on, {@code false} will turn it off.
-     * @throws IllegalStateException if called from a non-Looper thread.
+     * @throws IllegalStateException if called from a non-Looper thread with {@code autoRefresh} {@code true}.
```
Comment (cmelchior) [inline realm/realm-library/src/main/java/io/realm/BaseRealm.java]: Wouldn't it make more sense to just throw an IllegalStateException always? This behavior seems a bit weird?

Comment (zaki50) [inline realm/realm-library/src/main/java/io/realm/BaseRealm.java]: @cmelchior done. review again please

**Comment by zaki50** [comment]
updated the title

**Comment by cmelchior** [comment]
👍

**Comment by beeender** [comment]
:+1:

**Pull request merged**